src/Scripts/Ecs/Core/Entity.gd

```diff
@@ -8,9 +8,9 @@ var y:int
 # type (as a string) => instance
 var _components = {}
 
-func _init(x, y):
-	self.x = x
-	self.y = y
+func _init(_x, _y):
```

Comment:
This looks messed up to me. Member variables (class private variables) should be `_` prefixed, not method parameters.